### PR TITLE
feat(schema): Add schema generation

### PR DESCRIPTION
**Pull request by simar7**

### CODE REVIEW

File: pkg/rego/schemas/builder.go [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+
+type builder struct {
+	schema RawSchema
+}
+
+func Build() (*RawSchema, error) {
+
+	b := newBuilder()
+
+	inputValue := reflect.ValueOf(state.State{})
```
Comment (simar7) [inline pkg/rego/schemas/builder.go:33]: @nikpivkin `state.State{}` comes from defsec today and is shared across trivy-iac and trivy-aws as well. It's one of those dependencies which is hard to get away from as it's shared between all of the components.

File: pkg/rules/register.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package rules
+
+import (
+	"github.com/aquasecurity/defsec/pkg/framework"
+	"github.com/aquasecurity/defsec/pkg/scan"
+	"github.com/aquasecurity/trivy-iac/internal/rules"
+	"github.com/aquasecurity/trivy-policies/pkg/types"
+)
+
+func Register(rule scan.Rule, f scan.CheckFunc) types.RegisteredRule {
```
Comment (nikpivkin) [inline pkg/rules/register.go:10]: This function is used to register Go rules. For example, https://github.com/aquasecurity/trivy-policies/blob/main/rules/cloud/policies/aws/s3/block_public_acls.go#L11.

**Comment by nikpivkin** [comment]
@simar7 
If we have moved the embed and rego packages to trivy-iac, we should use them. For now, the packages from trivy-policies are used.
```go
"github.com/aquasecurity/trivy-policies/pkg/rego"
"github.com/aquasecurity/trivy-policies/pkg/rego/embed"
```

File: pkg/rules/register.go (hunk 2 above)
Comment (simar7) [reply pkg/rules/register.go:10]: > Maybe we should make Go rules in trivy-policies public and register them with trivy-iac?

Yes that's the idea. I was planning for that happen via https://github.com/aquasecurity/trivy-iac/blob/335c78327d3e3062b86e488b191adf13ded61c21/pkg/rego/embed.go#L9 where the rules are imported.

But it seems a couple of tests are failing at the moment. Upon investigating them it seems they don't seem to be registered Go rules, so an import somewhere is missing (which would trigger the registration of Go rules).

Comment (nikpivkin) [reply pkg/rules/register.go:10]: @simar7  When importing rules, they are saved in the rules registry, which is located in trivy-policies. Therefore, importing rules is not enough. https://github.com/aquasecurity/trivy-policies/blob/main/internal/rules/register.go#L22

**Review by nikpivkin — APPROVED** [review]
Tested with Trivy.

**Pull request merged**